code/game/machinery/vending/cm_vending.dm

```diff
@@ -902,108 +873,214 @@ IN_USE						used for vending/denying
 	. = ..()
 	populate_product_list(1.2)
 
+	for (var/i in 1 to length(listed_products))
+		// initial item count setup
+		var/item_name = listed_products[i][1]
+		var/initial_count = listed_products[i][2]
+		initial_product_count[item_name] = initial_count
+
+		// icon setup
+		var/typepath = listed_products[i][3]
+
+		var/icon_ref = null
+		var/icon_state = null
+		var/desc = ""
+		var/icon/r = null
+
+		if (ispath(typepath, /obj/effect/essentials_set))
+			var/obj/effect/essentials_set/I = new typepath()
+			var/list/spawned_list = I.spawned_gear_list
+			if(LAZYLEN(spawned_list))
+				var/obj/item/target = spawned_list[1]
+				icon_ref = initial(target.icon)
+				icon_state = initial(target.icon_state)
+				desc = initial(target.desc)
+				var/target_obj = new target()
+				r = getFlatIcon(target_obj)
+				qdel(target_obj)
+		else if (ispath(typepath, /obj/item))
+			var/obj/item/I = typepath
+			desc = initial(I.desc)
+			var/map_decor = initial(I.map_specific_decoration)
+			if (map_decor)
+				icon_state = initial(I.icon_state)
+				icon_ref = "icons/obj/items/weapons/guns/guns_by_map/classic/guns_obj.dmi"
+				r = icon(icon_ref, icon_state, SOUTH, 1)
+			else
+				var/target_obj = new I()
+				r = getFlatIcon(target_obj)
+				qdel(target_obj)
+
+		if (ispath(typepath, /obj/item/ammo_box/magazine))
+			var/obj/item/ammo_box/magazine/target = new typepath()
+			var/obj/item/magazine = target.magazine_type
+			desc = "A magazine box containing: [initial(magazine.desc)]"
```

Comment:
is this because they have no descs or?

Reply:
It's because it was just displaying the magazine description, was just a way to clarify that it was a box containing something

Reply:
just removed this